### PR TITLE
feat(chore): add test execution

### PR DESCRIPTION
**Pull request by mnlfischer**
## Description

Adding Vitest and a new action for automated testing. This PR only adds one component test for testing reasons.

## Checklist

- [ ] Change increases quality
- [ ] Change is validated by tests
- [ ] Change is readable and easy to understand
- [ ] Documentation is updated
- [ ] Security impact has been considered
- [ ] Changes validated in runtime

### CODE REVIEW

File: src/components/alert/alert.test.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { describe, expect, it } from "vitest";
+import { getByText, render } from "@testing-library/react";
+import React from "react";
+import { Alert } from "./alert";
+
+describe("Alert", () => {
+    it("renders an alert with a title and children", () => {
+        // ARRANGE
+        const { getByRole } = render(
```
Comment (pallendes) [inline src/components/alert/alert.test.tsx:9]: Using `screen.getByRole` is recommended by the creators, although it doesn't make a substantial difference 😄

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    pull_request_target:
+        types: [opened, reopened]
+
+jobs:
+    run-tests:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v3
+              with:
+                  persist-credentials: false # <--- this
```
Comment (pallendes) [inline .github/workflows/test.yaml:13]: What does that comment mean 🤔 ?

Comment (mnlfischer) [reply .github/workflows/test.yaml:13]: tbh I copied it from all other actions. I will remove it for all of them :D

File: src/components/alert/alert.test.tsx (hunk 1 above)
Comment (mnlfischer) [reply src/components/alert/alert.test.tsx:9]: I will change it to match the docs! Thanks for bringing that up. I run through different examples to get the additional "toBeInTheDocument()" functions to work and I guess I messed it up.

**Review by pallendes — APPROVED** [review]
lgtm 🎉

**Pull request merged**